From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: violetc <58360096+s-yh-china@users.noreply.github.com>
Date: Fri, 17 Mar 2023 15:57:08 +0800
Subject: [PATCH] Leaves Configurable Level Vanilla Random

Biquaternions note:
This changes have been validated with a vanilla jar for 26.1.2

diff --git a/net/minecraft/world/entity/item/PrimedTnt.java b/net/minecraft/world/entity/item/PrimedTnt.java
index 7970c6f9189059e8fc0391709159ef395c56708c..6228e90a968fabce7104d32cca9486903e945e3c 100644
--- a/net/minecraft/world/entity/item/PrimedTnt.java
+++ b/net/minecraft/world/entity/item/PrimedTnt.java
@@ -69,7 +69,10 @@ public class PrimedTnt extends Entity implements TraceableEntity {
     public PrimedTnt(final Level level, final double x, final double y, final double z, final @Nullable LivingEntity owner) {
         this(EntityType.TNT, level);
         this.setPos(x, y, z);
-        double rot = this.getRandom().nextDouble() * (float) (Math.PI * 2); // Paper - Don't use level random in entity constructors
+        // Chlorophyll start - Toggle Entity Level Randoms
+        net.minecraft.util.RandomSource random = me.biquaternions.chlorophyll.ChlorophyllConfig.TOGGLES.PAPER.PREVENT_LEVEL_CONSTRUCTOR_RANDOM ? this.random : level.random;
+        double rot = random.nextDouble() * (float) (Math.PI * 2); // Paper - Don't use level random in entity constructors
+        // Chlorophyll end - Toggle Entity Level Randoms
         this.setDeltaMovement(-Math.sin(rot) * 0.02, 0.2F, -Math.cos(rot) * 0.02);
         this.setFuse(80);
         this.xo = x;
